LambdaSource/Scotty_TableSlotUpdater/lambdaHandler.py

```diff
@@ -1,10 +1,98 @@
 #! python3
 import boto3
-import argparse
-
+import time
 from pprint import pprint
 
 
+def updateBot(lex, updatedSlotVersion):
+    bot = lex.get_bot(
+        name='Scotty',
+        versionOrAlias="$LATEST"
+    )
+    intentVersion = []
+    intentName = []
+    for intent in bot['intents']:
+        newIntent = lex.get_intent(name=intent['intentName'], version="$LATEST")
+        for i in range(len(newIntent['slots'])):
+            if "table" == newIntent['slots'][i]['name']:
+                newIntent['slots'][i]['slotTypeVersion'] = updatedSlotVersion
+                putIntent(lex, newIntent)
+                version = publishIntent(lex, newIntent)
+                intentVersion.append({'intentName': intent['intentName'], 'intentVersion': version})
+                intentName.append(intent['intentName'])
+
+        if intent['intentName'] not in intentName:
+            intentVersion.append({'intentName': intent['intentName'], 'intentVersion': intent['intentVersion']})
```

Comment:
You could probably get rid of this if statement, and the extra intentName list by doing something like this:

```updated_intent = {'intentName': intent['intentName'], 'intentVersion': intent['intentVersion']}```

Then, where you determine the intent has a table slot, you just update the version and then finally append the updated_intent to the intentVersion list, something like this:

```Python
for intent in bot['intents']:
    updated_intent = {'intentName': intent['intentName'], 'intentVersion': intent['intentVersion']}
    newIntent = lex.get_intent(name=intent['intentName'], version="$LATEST")
    for slot in newIntent['slots']:
        if slot['name'] == "table":
            slot['slotTypeVersion'] = updatedSlotVersion
            putIntent(lex, newIntent)
            version = publishIntent(lex, newIntent)
            updated_intent['intentVersion'] = version
            break
    intentVersion.append(updated_intent)
```